### PR TITLE
[Quant][Feature] Support online block_fp8 quant for dense and MoE models

### PR DESCRIPTION
**Pull request by EdalatiAli**
## Purpose
Add support for online block FP8 quantization (`--quantization block_fp8`), enabling BF16/FP16 models to be dynamically quantized to block-wise FP8 (with weight_block_size=[128, 128]) at load time — for both linear layers and MoE layers. Previously, block-wise FP8 quantization (weight_block_size) was only supported for pre-quantized (serialized) checkpoints.

## Usage
```
# Serve any BF16 model with online MXFP8 quantization
vllm serve <model> --quantization block_fp8

# Or with layer skipping via config.json:
# "quantization_config": {
#     "quant_method": "block_fp8",
#     "modules_to_not_convert": ["lm_head", "mlp.gate"]
# }
```

## Test Plan
E2E tests for both a dense model (Qwen/Qwen3-0.6B) and a MoE model (Qwen/Qwen3-30B-A3B): logprobs comparison against BF16 baseline + generation smoke test.

`python -m pytest tests/models/quantization/test_block_fp8.py -v
`

### Accuracy
```bash
lm_eval \
  --model vllm \
  --trust_remote_code \
  --model_args pretrained=Qwen/Qwen3-30B-A3B,quantization=block_fp8,enforce_eager=True \
  --tasks gsm8k,mmlu_pro \
  --batch_size auto
  ```
  
| Task | BF16 | Block FP8 |
|-----------|------|-------|
| GSM8K  (flexible-extract)   |   85.6   |   84.6    |
| GSM8K  (strict-match)   |   89.7   |   88.6    |
| MMLU-Pro  |   69.3  |    68.9   |

### Performance
```bash
vllm bench throughput --model Qwen/Qwen3-30B-A3B \
  --tensor-parallel-size 1 \
  --trust-remote-code \
  --async-scheduling \
  --backend vllm \
  --dataset-name random \
  --random-prefix-len 0 \
  --random-input-len 1024 \
  --random-output-len 1024 \
  --max-num-seqs 128 \
  --num-prompts 512 \
  --quantization block_fp8    ## Remove for running the BF16 model
```

#### BF16 performance
Throughput: 7.77 requests/s, 15910.87 total tokens/s, 7955.43 output tokens/s

#### Block FP8 performance
Throughput: 9.14 requests/s, 18719.65 total tokens/s, 9359.83 output tokens/s

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/fp8.py [hunk 1]
```diff
@@ -540,27 +547,50 @@ def process_weights_after_loading(self, layer: Module) -> None:
...
+            )
+
+            qweight, weight_scale_inv = process_fp8_weight_block_strategy(
+                qweight, weight_scale_inv
+            )
+
+            replace_parameter(layer, "weight", qweight.data)
+            layer.weight_scale_inv = torch.nn.Parameter(
+                weight_scale_inv.data, requires_grad=False
+            )
```
[the remark this answers is not in the document]
Comment (EdalatiAli) [reply vllm/model_executor/layers/quantization/fp8.py]: Addressed

File: vllm/model_executor/layers/quantization/fp8.py [hunk 2]
```diff
@@ -1006,29 +1056,70 @@ def create_weights(
...
+                w2_scale_list.append(s2)
+
+            w13 = torch.stack(w13_list)
+            w13_scale = torch.stack(w13_scale_list)
+            w2 = torch.stack(w2_list)
+            w2_scale = torch.stack(w2_scale_list)
+
+            if current_platform.is_fp8_fnuz():
+                w13, w13_scale, _ = normalize_e4m3fn_to_e4m3fnuz(w13, w13_scale)
+                w2, w2_scale, _ = normalize_e4m3fn_to_e4m3fnuz(w2, w2_scale)
```
[the remark this answers is not in the document]
Comment (EdalatiAli) [reply vllm/model_executor/layers/quantization/fp8.py]: Addressed
